### PR TITLE
Exitpoll: Refactoring interactions

### PR DESCRIPTION
**Pull request by AtifMahmud**
# Description

- When `Right Controller Pointer` or `Left Controller Pointer` is selected as `Exit Poll Pointer Type`, we can provide hand tracking support if hand tracking is active in scene for the following platforms:
    - **Meta:** Users will need to pinch with index finger to “click”
- Added support for configuring error message to display on panel
- Removed duplicate code
- Updated functionality to easily configure interaction type, i.e. trigger, pinch, gaze
- In hand/controller mode, if controllers and/or hands unavailable for more than 35 seconds, fallback to `HMDPointer`

Height Task ID(s) (If applicable): T-7637

Details on testing: https://www.notion.so/cognitive3d/T-7637-Use-Hands-to-answer-ExitPoll1-03a00939d1c749daa0526cc2a5403780?pvs=4

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist

- [x] My changes will not interrupt or disrupt automated build processes
- [x] I have checked all outgoing links (i.e. to documentation) for validity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have assigned this PR to myself in GitHub
- [x] I have assigned and notified Reviewers for this PR

### CODE REVIEW

File: Editor/ExitPollHolderInspector.cs [hunk 1]
```diff
@@ -195,29 +195,18 @@ public override void OnInspectorGUI()
 
         private string GetPointerDescription(ExitPollParameters parameters)
         {
-            string thingToSpawn = "";
-            string howToAttach = "";
             if (parameters.PointerType == ExitPoll.PointerType.LeftControllerPointer)
             {
-                thingToSpawn = "Spawn ExitPollControllerPointer";
-                howToAttach = " and attach to Left Controller";
+                return "Users will interact with the buttons by using the left controller and/or left hand, if available.";
```
Comment (AtifMahmud) [inline Editor/ExitPollHolderInspector.cs:200]: For reviewer: Rewording the text users see on the inspector.

File: Runtime/ExitPoll/ExitPoll.cs [hunk 2]
```diff
@@ -54,12 +53,6 @@ public class ExitPollSetJsonEntryAnswer
     //static class for requesting exitpoll question sets with multiple panels
     public static class ExitPoll
     {
-        public enum PointerSource
```
Comment (AtifMahmud) [inline Runtime/ExitPoll/ExitPoll.cs:57]: This was unused. Thus removing.

File: Runtime/ExitPoll/ExitPoll.cs [hunk 3]
```diff
@@ -201,6 +203,48 @@ public void BeginExitPoll(ExitPollParameters parameters)
...
+        {
+            panelProperties.Clear();
+            if (CurrentExitPollPanel != null)
+            {
+                CurrentExitPollPanel.CloseError(timeToWait);
+            }
+            OnPanelError();
+        }
+
+        private void Cognitive3D_Manager_OnUpdate(float deltaTime)
```
Comment (AtifMahmud) [inline Runtime/ExitPoll/ExitPoll.cs:220]: If controllers and/or hands are unavailable for over a certain time, fallback to HMD

File: Runtime/ExitPoll/ExitPoll.cs [hunk 4]
```diff
@@ -212,39 +256,19 @@ private void SetupControllerAsPointer(bool isRight)
             Transform t = null;
             if (pointerInstance != null)
             {
-                if (isRight)
+                GameplayReferences.controllerPointer = pointerInstance;
```
Comment (AtifMahmud) [inline Runtime/ExitPoll/ExitPoll.cs:259]: Removed duplicate code here

File: Runtime/ExitPoll/ExitPoll.cs [hunk 5]
```diff
@@ -418,7 +452,7 @@ public void OnPanelError()
         {
             CurrentExitPollPanel = null;
             Cleanup(false);
-            Cognitive3D.Util.logDebug("Exit poll OnPanelError - HMD is null, manually closing question set or new exit poll while one is active");
+            Util.logDebug("Exit poll OnPanelError - HMD is null, manually closing question set or new exit poll while one is active");
```
Comment (AtifMahmud) [inline Runtime/ExitPoll/ExitPoll.cs:455]: Namespace cleanups

File: Runtime/Scripts/ControllerPointer.cs [hunk 6]
```diff
@@ -11,72 +11,123 @@ namespace Cognitive3D
...
+        [HideInInspector]
         public Material DefaultPointerMat;
-        Material FocusPointerMat;
-        private bool focused;
-        public bool DisplayLineRenderer = true;
-        public LineRenderer LineRendererOverride;
-        Vector3[] pointsArray;
 
-        LineRenderer lr;
+        [HideInInspector]
```
Comment (AtifMahmud) [inline Runtime/Scripts/ControllerPointer.cs:17]: Updated script to easily configure more hand tracking in future. Pinch gesture, custom activation, etc.

File: Runtime/Scripts/GameplayReferences.cs [hunk 7]
```diff
@@ -383,15 +361,60 @@ public static Transform HMD
...
+
         //dynamic objects set as controllers call 'set controller' on enable passing a reference to that transform. input device isn't guaranteed to be valid at this point
 
         static Transform[] controllerTransforms = new Transform[2];
         static InputDevice[] controllerDevices = new InputDevice[2];
 
+        public static float rightTriggerValue;
+        public static float leftTriggerValue;
+
+        //updates controller and hmd inputdevices to call events when states change
```
Comment (AtifMahmud) [inline Runtime/Scripts/GameplayReferences.cs:394]: Added option to retrieve trigger value from GameplayReferences. Will be helpful to have it one place.

File: Runtime/Scripts/VirtualButton.cs [hunk 8]
```diff
@@ -122,35 +99,50 @@ protected virtual void LateUpdate()
             if (isEnabled && !isSelected)
             {
                 if (!gameObject.activeInHierarchy) { return; }
```
Comment (AtifMahmud) [inline Runtime/Scripts/VirtualButton.cs:101]: Cleanup

**Review by calderarchinuk — APPROVED** [review]

**Pull request merged**